java/ql/src/experimental/Security/CWE/CWE-094/SpringViewManipulationLib.qll

```diff
@@ -1,8 +1,23 @@
+/**
+ * Provides classes for reasoning about Spring View Manipulation vulnerabilities
+ */
+
 import java
 import semmle.code.java.dataflow.FlowSources
 import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.frameworks.spring.Spring
 import SpringFrameworkLib
 
+predicate thymeleafIsUsed() {
+  exists(Pom p, Artifact a | a = p.getADependency().getArtifact() |
+    a.getValue() =
+      ["spring-boot-starter-thymeleaf", "thymeleaf-spring4", "springmvc-xml-thymeleaf",
+          "thymeleaf-spring5"]
+  )
+  or
+  exists(SpringBean b | b.getClassNameRaw().matches("org.thymeleaf.spring4"))
```

Comment:
Is this meant to match `ThymeleafViewResolver` beans? if so do you need to account for both `org.thymeleaf.spring5.view.ThymeleafViewResolver` and `org.thymeleaf.spring4.view.ThymeleafViewResolver`?